### DISCUSSION TITLE
Reconnecting to a dapp after disconnect doesn't work

### DISCUSSION THREAD

**Discussion by fareed945** [WalletConnect/walletconnect-monorepo#3246] [Developer Support] [answered]
I'm using the [https://react-app.walletconnect.com/] dapp for testing my wallet using the version 2.9.1. 
These are the steps that I follow, 
1) Take the uri from the dapp and connect to it by pairing with it, this works all good.
2) I disconnect from the dapp with the following code in the wallet 
```
export async function disconnectSession() {
  let sessions: any = getActiveSessions();
  sessions.forEach(async ({ topic }) => {
    await web3wallet.disconnectSession({
      topic,
      reason: walletconnectUtils.getSdkError("USER_DISCONNECTED"),
    });
  });

  // Disconnect all pairings and sessions
  let pairings = getActivePairings();

  pairings.forEach(async ({ topic }) => {
    await web3wallet.disconnectSession({
      topic,
      reason: walletconnectUtils.getSdkError("USER_DISCONNECTED"),
    });

  });
}
```
3) Now I try reconnecting to the dapp with a new pairing, with a different uri, but end up with errors on the dapp side
```
_app-34b8e6c7f6529fc3.js:163 Uncaught (in promise) Error: No matching key. history: 1689964818809219
    at dB.getRecord (_app-34b8e6c7f6529fc3.js:163:1196333)
    at dB.get (_app-34b8e6c7f6529fc3.js:163:1194964)
```
Please help !!

**Comment by glitch-txs** [comment, marked as answer]
Aswered [here](https://github.com/orgs/WalletConnect/discussions/3291). Seems to be an issue on the Wallet side.
